testdata/extern_modules/extern-checksum-ebpf.c

```diff
@@ -0,0 +1,37 @@
+#include  "test.h"
+
+/*
+ * This file implements sample C extern function. It contains definition of the following C extern function:
+ *
+ * bool verify_ipv4_checksum(in IPv4_h iphdr)
+ *
+ */
+
+
+/**
+ * This function implements method to verify IP checksum. It shows how to write C extern function for P4 program.
+ * @param iphdr Structure representing IP header. The IP header is generated by the P4 compiler.
+ * @return True if checksum is correct.
```

Comment:
The IP header is not really generated by the compiler; it is the C representation of a user-defined datatype (which much be present in the p4 source program which is being compiled) which is supposed to model an IPv4 header without options. I think that this should be part of the documentation of this function.

Reply:
The `IPv4_h` is taken from `test.h`, which is generated by the compiler. I've modified comment to be more concrete.